lib/moon/design/dropdown/options.ex

```diff
@@ -84,4 +88,68 @@ defmodule Moon.Design.Dropdown.Options do
     </div>
     """
   end
+
+  defp get_options(assigns), do: assigns[:option] || assigns[:titles]
+
+  defp get_max_h_classes(max_length, list, "sm") when is_list(list) and is_integer(max_length) do
+    if length(list) > max_length do
+      case max_length do
+        1 -> ~w(dropdown-options-scrollable max-h-8)
+        2 -> ~w(dropdown-options-scrollable max-h-16)
+        3 -> ~w(dropdown-options-scrollable max-h-24)
+        4 -> ~w(dropdown-options-scrollable max-h-32)
+        5 -> ~w(dropdown-options-scrollable max-h-40)
+        _ -> ~w(dropdown-options-scrollable max-h-48)
+      end
+    else
+      []
+    end
```

Comment:
A bit cleaner version that does the same thing:

```suggestion
    case max_length do
      x when length(list) > x -> []
      1 -> ~w(dropdown-options-scrollable max-h-8)
      2 -> ~w(dropdown-options-scrollable max-h-16)
      3 -> ~w(dropdown-options-scrollable max-h-24)
      4 -> ~w(dropdown-options-scrollable max-h-32)
      5 -> ~w(dropdown-options-scrollable max-h-40)
      _ -> ~w(dropdown-options-scrollable max-h-48)
    end
```

Reply:
I personally think merging the `get_max_h_classes` is slightly better:
- By eliminating the need for duplication
- Making easier to compare how different sizes are configured

```elixir
    case {size, max_length} do
      {_, x} when length(list) > x -> []
      # small
      {"sm", 1} -> ~w(dropdown-options-scrollable max-h-8)
      {"sm", 2} -> ~w(dropdown-options-scrollable max-h-16)
      {"sm", 3} -> ~w(dropdown-options-scrollable max-h-24)
      {"sm", 4} -> ~w(dropdown-options-scrollable max-h-32)
      {"sm", 5} -> ~w(dropdown-options-scrollable max-h-40)
      {"sm", _} -> ~w(dropdown-options-scrollable max-h-48)
      # medium
      {"md", 1} -> ~w(dropdown-options-scrollable max-h-10)
      {"md", 2} -> ~w(dropdown-options-scrollable max-h-20)
      ...
      # large
      {"lg", 1} -> ~w(dropdown-options-scrollable max-h-12)
      ...
    end

Reply:
Thank you.
Regarding the first comment, I completely agree. It's better to merge the two conditions.

Concerning the second one, I'm not so sure. As a design, I honestly prefer the expressive design of functions, keeping them separate for improved understanding.
`a("md")`
`a("lg")`

is much more clear than `a(size) do: case ...`

Surely will make shorter file, but at the same time, it create this long case matching. I will think about it 👍